### PR TITLE
Add vmla_* / vmlal_*

### PR DESCRIPTION
**Pull request by makotokato**
Add Vector multiply accumulate intrinsic.

### CODE REVIEW

**Comment by Amanieu** [comment]
VMLA is *not* a fused multiply-add. It involves an intermediate rounding step just like if you use two separate operations. Only VFMA is a fused multiply-add.

**Comment by Amanieu** [comment]
Ping. The `vmla_*` functions needs to use `simd_add` and `simd_mul`. The `vfma_*` functions need to use `simd_fma`.

File: crates/core_arch/src/aarch64/neon/mod.rs [hunk 1]
```diff
@@ -971,6 +971,40 @@ pub unsafe fn vaddvq_u64(a: uint64x2_t) -> u64 {
...
+    simd_fma(b, c, a)
+}
+
+/* FIXME: simd_fma doesn't support float64x1_t`
+/// Vector multiply accumulate
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(fmla))]
+pub unsafe fn vmla_f64(a: float64x1_t, b: float64x1_t, c: float64x1_t) -> float64x1_t {
+    simd_fma(b, c, a)
```
Comment (Amanieu) [inline crates/core_arch/src/aarch64/neon/mod.rs:988]: You can use the `mul_add` method on `f64`.

Comment (makotokato) [reply crates/core_arch/src/aarch64/neon/mod.rs:988]: Sorry for delay an thank you for review. I will update this PR.

**Comment by makotokato** [comment]
unrelated CI is failure due to OOM...

**Comment by Amanieu** [comment]
CI should be fixed now.

However you still haven't fixed the definitions of the functions: 
- `vfma_f{32,64}` should always do a fused-multiply add. (`vfma` on ARM, `fmadd`/`fmla` on AArch64`)
- `vmla_f{32,64}` should always do a separate multiple and add. (`vmla` on ARM, `fmul` + `fadd` on AArch64).

The `vfma_f{32,64}` functions are also available on 32-bit ARM if the vfpv4 feature is enabled.

Have a look at this: https://godbolt.org/z/34Wo5n

**Pull request closed without merging**